src/main/kotlin/graphics/scenery/Camera.kt

```diff
@@ -336,14 +336,30 @@ open class Camera : DefaultNode("Camera"), HasRenderable, HasMaterial, HasCustom
      *
      * It will be shown for [duration] milliseconds, with a default of 3000.
      */
-    @JvmOverloads fun showMessage(message: String, distance: Float = 0.75f, size: Float = 0.05f, messageColor: Vector4f = Vector4f(1.0f), backgroundColor: Vector4f = Vector4f(0.0f), duration: Int = 3000) {
+    @JvmOverloads
+    fun showMessage(
+        message: String,
+        distance: Float = 0.75f,
+        size: Float = 0.05f,
+        messageColor: Vector4f = Vector4f(1.0f),
+        backgroundColor: Vector4f = Vector4f(0.0f),
+        duration: Int = 3000,
+        centered: Boolean = false
+    ) {
         val tb = TextBoard()
         tb.fontColor = messageColor
         tb.backgroundColor = backgroundColor
         tb.text = message
+        // Calculate the offset for text centering based on message length.
+        // The second divisor was eyeballed and might need adjustment.
```

Comment:
if you do it like this https://github.com/moreApi/microscenery/blob/52b2e5156609f8489c9fb053d508db9ec12170d2/frontend/src/main/kotlin/microscenery/VRUI/Gui3D/TextBox.kt#L57 you dont need to guess :)

Reply:
Good suggestion, I changed it 👍

Reply:
sorry I should have been clearer. the part before the linked line is also relevant. I basically check for the right most point of the geometry and then apply the scale.

Reply:
this has been fixed now